lib/asset/AssetJavascriptGenerator.js

```diff
@@ -42,7 +42,11 @@ class AssetJavascriptGenerator extends Generator {
 	generate(module, { chunkGraph, runtimeTemplate, runtimeRequirements }) {
 		const filename = module.nameForCondition();
 		const { assetModuleFilename } = runtimeTemplate.outputOptions;
-		const url = this.mainTemplate.getAssetPath(assetModuleFilename, {
+		const moduleFilename =
```

Comment:
This seems unnessecary as `mainTemplate.getAssetPath` already supports passing a function.

Reply:
Yep, but if we pass moduleFilename as a function, we'll get `module: NormalModule` as an argument at `assetFilename` because `filename` is `module.filename`.

Reply:
You should get an object `{ module, filename, chunkGraph }` and that's fine.

Reply:
OK, so I'll change below.

```js
assetModuleFilename: ({ filename }) => {
  if (/.png$/.test(filename)) {
    return "images/success-png[ext]";
  }
  if (/.svg$/.test(filename)) {
    return "images/success-svg[ext]";
  }
  return "images/failure[ext]";
}
```